app/controllers/topics_controller.rb

```diff
@@ -16,6 +16,24 @@ def create
     end
   end
 
+  def show
+    @topic = Topic.find_by_id(params[:id])
+
+    if @topic.nil?
+      redirect_to topics_path
+    end
+  end
+
+  def edit
+    @topic = Topic.find_by_id(params[:id])
+  end
+
+  def update
+    @topic = Topic.find_by_id(params[:id])
+    @topic.update!(topic_params)
```

Comment:
So, if the user makes a mistake this controller action will just blow up and present the user with the 500 page. 
What about all the hard work you did to validate the Topic?
Take a look at your `create` action and give this another go.